oncopipe/oncopipe/__init__.py

```diff
@@ -206,25 +191,79 @@ def coerce_namedlist_to_string(obj):
     # Or you are symlinking a directory to a specific location
     else:
         dest_dir, dest_file = os.path.split(dest)
-    os.makedirs(dest_dir, exist_ok=True)
-
+    os.makedirs(os.path.abspath(dest_dir), exist_ok=True)
     dest = os.path.join(dest_dir, dest_file)
+    
+    return src, dest
+
+def compare_links(src, dest, overwrite):
+    # Check if the destination file exists and is a symlink
     if os.path.lexists(dest) and os.path.islink(dest):
-        if os.path.realpath(src) == os.path.realpath(dest):
-            return
-        elif overwrite:
+        # If the destination link exists and has the same src, exit
+        if src == os.readlink(dest) or overwrite:
             os.remove(dest)
     assert not os.path.exists(dest), (
         "Symbolic link already exists but points elsewhere: \n"
```

Comment:
One issue I've run into here on testing is that if the destination file exists but is a file and not a symlink, the error message doesn't work because there's no link to read. Can you think of a solution? It still generates an error, but not a helpful one.

Reply:
Good catch! I would add an assertion before the existing one to handle that edge case for a more informative error message:

```python
    # Raise error if the file exists and isn't a symbolic link
    assert not (os.path.exists(dest) and not os.path.islink(dest)), (
        "Destination file already exists but isn't a symbolic link: \n"
        f"    Current: {dest} \n"
        f"    Attempted: {dest} -> {src}"
    )
    # Raise error if the file exists and is a symbolic link
    assert not (os.path.exists(dest) and os.path.islink(dest)), (
        "Symbolic link already exists but points elsewhere: \n"
        f"    Current: {dest} -> {os.readlink(dest)} \n"
        f"    Attempted: {dest} -> {src}"
    )
```